wpt-dashboard-stats.py

```diff
@@ -1,34 +1,72 @@
 #!/usr/bin/env python
-# Requirements: python-dateutil, numpy, requests
 
-from __future__ import print_function
-from collections import defaultdict, namedtuple
-import csv
 import dateutil.parser
-import json
-import numpy
-import re
 import requests
-import subprocess
 
-from wpt_common import CUTOFF, QUARTER_START, fetch_all_prs, wpt_git
+from csv_database import RunLatencyDB
+from wpt_common import fetch_all_prs, get_pr_latencies
 
-RUNS_URL='https://wpt.fyi/api/runs?max-count=100'
-CSV_FILE = 'wpt-dashboard-latency.csv'
+# 1000 because of https://github.com/w3c/wptdashboard/issues/524
+RUNS_URL='https://wpt.fyi/api/runs?max-count=1000'
+CSV_PATH_TEMPLATE = 'wpt-dashboard-{}-latency.csv'
 
+# If the runs expand to non-desktop platforms, those should be measured
+# separately, so assert that only desktop configurations exist.
+KNOWN_RUN_CONFIGS = set([
+    ('chrome', 'debian'),
+    ('chrome', 'linux'),
+    ('chrome', 'linux\n'),
+    ('edge', 'windows'),
+    ('firefox', 'debian'),
+    ('firefox', 'linux'),
+    ('safari', 'macos'),
+    ('safari', 'macOS'),
+])
 
-def get_latencies(prs, runs):
+
+def run_sha(run):
+    return run['revision']
+
+
+def run_date(run):
+    return dateutil.parser.parse(run['created_at'])
+
+
+def calculate_latencies(prs, runs):
     """For each PR, find the earliest run for each browser that included that PR,
     and calucate the latencies between the PR and the runs."""
-    latencies = {}
-    return latencies
+
+    for run in runs:
+        # If this assert fails, add to the known set if it's a desktop
+        # configuration, otherwise filter out the run.
+        assert (run['browser_name'], run['os_name']) in KNOWN_RUN_CONFIGS
+
+    browsers = list(set(run['browser_name'] for run in runs))
+    browsers.sort()
+
+    # Per-browser latencies:
+    for name in browsers:
+        browser_runs = [run for run in runs if run['browser_name'] == name]
+        latencies = get_pr_latencies(prs, events=browser_runs, event_sha=run_sha, event_date=run_date)
```

Comment:
The naming here is a bit strained, and in your scripts the CSV DB is called `latencies`